### PR TITLE
Add GitHub data analysis notebook

### LINKED ISSUE #16: Notebook to explore the issue/pr data for OSG repos
We have collected GitHub PR/Issue data in #13 and would like to further analyse the raw data to identify relevant fields/columns for calculating metrics.
(Related #6)

### LINKED ISSUE #956: Latest metrics for Jupyterhub

### PR DESCRIPTION
**Pull request by hemajv**
Resolves operate-first/support#956

### CODE REVIEW

**Comment by suppathak** [comment]
@hemajv , how can i have access to [review.nb](https://app.reviewnb.com/) for this repo, in order to review the notebook?

**Comment by hemajv** [comment]
<p>@suppathak thanks for pointing that out! I just installed the reviewnb app for this repo. Can you check if you are able to review it here https://app.reviewnb.com/open-services-group/metrics/pull/17/?</p>

**Comment by hemajv** [comment]
> @hemajv , how can i have access to [review.nb](https://app.reviewnb.com/) for this repo, in order to review the notebook?

@suppathak thanks for pointing that out! I just installed the reviewnb app for this repo, can you check if you are able to review it here: https://app.reviewnb.com/open-services-group/metrics/pull/17/?

**Comment by suppathak** [comment]
> @suppathak thanks for pointing that out! I just installed the reviewnb app for this repo, can you check if you are able to review it here: https://app.reviewnb.com/open-services-group/metrics/pull/17/?

Yup! its installed now. I can view it. Thank you. :smiley: :rocket:

**Comment by suppathak** [comment]
@hemajv , Great doc :clap: :clap:. :smiley:  I could walk-through the notebook quite clearly. It gives us a clear understanding of how to fetch the data for all the sub projects under different SIGs. 

However,  in the issue operate-first/support#956 , it talks about further analyzing the data to identify relevant field/columns to calculate the metrics. Are we planning to do that in the same notebook? or planning to create different PR's to do data analysis for different sub-projects?

**Review by Shreyanand — COMMENTED** [review]
/lgtm :100: The pre-commit doesn't seem very happy, any ideas why?

**Comment by hemajv** [comment]
> @hemajv , Great doc clap clap. smiley I could walk-through the notebook quite clearly. It gives us a clear understanding of how to fetch the data for all the sub projects under different SIGs.
> 
> However, in the issue operate-first/support#956 , it talks about further analyzing the data to identify relevant field/columns to calculate the metrics. Are we planning to do that in the same notebook? or planning to create different PR's to do data analysis for different sub-projects?

@suppathak yes exactly, we plan to have a separate notebook for that :+1:

**Comment by hemajv** [comment]
> /lgtm 100 The pre-commit doesn't seem very happy, any ideas why?

hmm the test seems to pass locally for me, @harshad16 pointed out that it should be related to https://github.com/operate-first/support/issues/673

File: notebooks/GitHub_Data_EDA.ipynb [hunk 1]
```diff
@@ -0,0 +1,4895 @@
+{
```
Comment (chauhankaranraj) [inline notebooks/GitHub_Data_EDA.ipynb:1]: <p>Small nit, should we clarify here that <code>mi</code> is the tool that does the github data fetching, and <code>mi-scheduler</code> is the tool that sets up scheduled pipelines for data collection? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-services-group/metrics/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (hemajv) [reply notebooks/GitHub_Data_EDA.ipynb:1]: <p>good point! just updated :)</p>

**Review by suppathak — COMMENTED** [review]
/lgtm
/approve

Awsm doc :clap: :clap:

**Pull request merged**